From aa6bf73d4ad828bb87b7f02a0b17e9f98360bb1b Mon Sep 17 00:00:00 2001
From: Peter Eckersley <pde@eff.org>
Date: Mon, 21 Dec 2015 19:57:12 -0800
Subject: [PATCH 1/3] Only test permission failures if we're not root

or, more generally, if we're on a system where permissions are being enforced

Closes: #1979
---
 letsencrypt/plugins/webroot_test.py | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/letsencrypt/plugins/webroot_test.py b/letsencrypt/plugins/webroot_test.py
index 9f5b6bb..07e41e0 100644
--- a/letsencrypt/plugins/webroot_test.py
+++ b/letsencrypt/plugins/webroot_test.py
@@ -66,8 +66,17 @@ class AuthenticatorTest(unittest.TestCase):
 
     def test_prepare_reraises_other_errors(self):
         self.auth.full_path = os.path.join(self.path, "null")
+        permission_canary = os.path.join(self.path, "rnd")
+        f = open(permission_canary, "w")
+        f.write("thingimy")
+        f.close()
         os.chmod(self.path, 0o000)
-        self.assertRaises(errors.PluginError, self.auth.prepare)
+        try:
+            open(permission_canary, "r")
+            print("Warning, running tests as root skips permissions tests...")
+        except IOError:
+            # ok, permissions work, test away...
+            self.assertRaises(errors.PluginError, self.auth.prepare)
         os.chmod(self.path, 0o700)
 
     @mock.patch("letsencrypt.plugins.webroot.os.chown")
-- 
2.6.4

From e41339cda8e8d091f0bc7babbdd9098c7d17a1f7 Mon Sep 17 00:00:00 2001
From: Peter Eckersley <pde@eff.org>
Date: Mon, 21 Dec 2015 20:01:28 -0800
Subject: [PATCH 2/3] Keep lint happy

(But what about py3?)
---
 letsencrypt/plugins/webroot_test.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/letsencrypt/plugins/webroot_test.py b/letsencrypt/plugins/webroot_test.py
index 07e41e0..137a267 100644
--- a/letsencrypt/plugins/webroot_test.py
+++ b/letsencrypt/plugins/webroot_test.py
@@ -73,7 +73,7 @@ class AuthenticatorTest(unittest.TestCase):
         os.chmod(self.path, 0o000)
         try:
             open(permission_canary, "r")
-            print("Warning, running tests as root skips permissions tests...")
+            print "Warning, running tests as root skips permissions tests..."
         except IOError:
             # ok, permissions work, test away...
             self.assertRaises(errors.PluginError, self.auth.prepare)
-- 
2.6.4

From f5cf58f42ef0704a9b4ddf122310527764d727ba Mon Sep 17 00:00:00 2001
From: Peter Eckersley <pde@eff.org>
Date: Tue, 22 Dec 2015 15:42:53 -0800
Subject: [PATCH 3/3] with .. open .. as   # definitely nicer

---
 letsencrypt/plugins/webroot_test.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/letsencrypt/plugins/webroot_test.py b/letsencrypt/plugins/webroot_test.py
index 137a267..defe939 100644
--- a/letsencrypt/plugins/webroot_test.py
+++ b/letsencrypt/plugins/webroot_test.py
@@ -67,9 +67,8 @@ class AuthenticatorTest(unittest.TestCase):
     def test_prepare_reraises_other_errors(self):
         self.auth.full_path = os.path.join(self.path, "null")
         permission_canary = os.path.join(self.path, "rnd")
-        f = open(permission_canary, "w")
-        f.write("thingimy")
-        f.close()
+        with open(permission_canary, "w") as f:
+            f.write("thingimy")
         os.chmod(self.path, 0o000)
         try:
             open(permission_canary, "r")
-- 
2.6.4

